src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int amountInStore = FruitStorage.fruitRepository.get(fruit);

```

Comment:
There's a risk of a `NullPointerException` here if `FruitStorage.fruitRepository.get(fruit)` returns `null` because the fruit doesn't exist in the repository yet. You should handle this case appropriately to avoid runtime exceptions.